src/add-remove.js

```diff
@@ -93,42 +93,48 @@ function AddTask(description) {
   taskArray.push(new Task(description));
   localStorage.setItem('taskArray', JSON.stringify(taskArray));
   addInput.value = '';
-  DisplayTask();
+  DisplayTask(taskArray);
+}
+
+function EditTesting(indexToUpdate, taskEdit) {
+  taskArray[indexToUpdate - 1].description = taskEdit;
+  localStorage.setItem('taskArray', JSON.stringify(taskArray));
+  DisplayTask(taskArray);
 }
 
 function EditTask(indexToUpdate, descriptionToUpdate) {
   // eslint-disable-next-line
   const taskEdit = prompt(`Editing '${descriptionToUpdate}' task`, descriptionToUpdate);
   if (taskEdit != null) {
-    taskArray[indexToUpdate - 1].description = taskEdit;
-    localStorage.setItem('taskArray', JSON.stringify(taskArray));
-    DisplayTask();
+    EditTesting(indexToUpdate, taskEdit);
   }
 }
 
 function DeleteTask(indexToDelete) {
   taskArray = taskArray.filter((task) => task.index !== indexToDelete);
   indexToAssign -= 1;
   UpdateIndexes();
-  DisplayTask();
+  DisplayTask(taskArray);
 }
 
-function DeleteAllCompletedTasks() {
-  if (localStorage.getItem('taskArray')) {
-    taskArray = JSON.parse(localStorage.getItem('taskArray'));
-  }
+function DeleteAllCompletedTasks(taskArray) {
+  // if (localStorage.getItem('taskArray')) {
+  //   taskArray = JSON.parse(localStorage.getItem('taskArray'));
+  // }
   // eslint-disable-next-line
-  taskArray = taskArray.filter((task) => task.completed != true);
+  const filteredArray = taskArray.filter((task) => task.completed != true);
```

Comment:
Mean reason for writing a unit test is to figure out when we have a bug in the actual function itself.

Example:
- You have a function called `DeleteAllCompletedTasks` in `add-remove.js`
- You wrote a test for this function in `add-remove.test.js`
- I went to `DeleteAllCompletedTasks` in `add-remove.js` to introduce a bug, I'm expecting the test to fail.

In your case;

Change true to false in this function, that is, instead of deleting a completed task, let's delete an incomplete task, the test case didn't catch that.

Reply:
I'm sorry bellom, I think I didn't understand this one, could you please explain a bit better what do you mean?

Reply:
@jmontanholi, the basic idea is to introduce a bug in the actual function.

So try to introduce a bug function by changing `task.completed != true` to `task.completed != false`, then run the test. 

The test suite still passed, the test suite should fail.

Does this make sense now.

Reply:
Yes, we'll take a look at it, thanks!